From 2605d477d95efe1cc2ce4895f88333def6b2371a Mon Sep 17 00:00:00 2001
From: Don <don.j.olmstead@gmail.com>
Date: Fri, 12 Nov 2021 11:09:37 -0800
Subject: [PATCH 2/4] Set RUNTIME_OUTPUT_DIRECTORY for test executables

Setting the target's PREFIX causes ninja to fail claiming multiple rules generate the same executable. The prefix is just a directory name so use RUNTIME_OUTPUT_DIRECTORY instead.
---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 43047aa..faaf424 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -379,7 +379,7 @@ foreach (TESTFILE IN LISTS HWY_TEST_FILES)
     target_link_libraries(${TESTNAME} hwy hwy_contrib hwy_test gtest gtest_main)
   endif()
   # Output test targets in the test directory.
-  set_target_properties(${TESTNAME} PROPERTIES PREFIX "tests/")
+  set_target_properties(${TESTNAME} PROPERTIES RUNTIME_OUTPUT_DIRECTORY "tests")
 
   if (HWY_EMSCRIPTEN)
     set_target_properties(${TESTNAME} PROPERTIES LINK_FLAGS "-s SINGLE_FILE=1")
-- 
2.33.0.windows.2

